rust/common/kafka/src/kafka_producer.rs

```diff
@@ -55,7 +55,22 @@ pub async fn create_kafka_producer(
     let api: FutureProducer<KafkaContext> =
         client_config.create_with_context(KafkaContext { liveness })?;
 
-    // TODO: ping the kafka brokers to confirm configuration is OK (copy capture)
+    // "Ping" the Kafka brokers by requesting metadata
+    match api
+        .client()
+        .fetch_metadata(None, std::time::Duration::from_secs(2))
```

Comment:
```suggestion
        .fetch_metadata(None, std::time::Duration::from_secs(15))
```

I think I'd bump this just to be extra safe. If the Kafka brokers were for some reason acting slow, or maybe the metadata request took a long time, I would hate for the process to crash-loop rather than wait it out.

Reply:
Updated, thank you!